src/OVAL/probes/unix/linux/auditdline_probe.c

```diff
@@ -0,0 +1,1077 @@
+			} else if (field == AUDIT_FILTERKEY) {
+				char *rkey, *ptr, *saved;
+				if (asprintf(&rkey, "%.*s", r->values[i],
+					     &r->buf[boffset]) < 0)
+					rkey = NULL;
+				boffset += r->values[i];
+				ptr = strtok_r(rkey, key_sep, &saved);
+				while (ptr) {
+					if (watch) {
+						field_value_len = snprintf(
+							NULL, 0, " -k %s", ptr);
+						char *audit_field = malloc(
+							field_value_len + 1);
+						snprintf(audit_field,
+							 field_value_len + 1,
+							 " -k %s", ptr);
+						audit_line_push(audit_line,
+								audit_field);
+						audit_line_add_filter_key(
+							audit_line, ptr);
+						free(audit_field);
+					} else {
+						field_value_len =
+							snprintf(NULL, 0,
+								 " -F key=%s",
+								 ptr);
+						char *audit_field = malloc(
+							field_value_len + 1);
+						snprintf(audit_field,
+							 field_value_len + 1,
+							 " -F key=%s", ptr);
+						audit_line_push(audit_line,
+								audit_field);
+						audit_line_add_filter_key(
+							audit_line, ptr);
+						free(audit_field);
+					}
+					ptr = strtok_r(NULL, key_sep, &saved);
+				}
+				free(rkey);
+			} else if (field == AUDIT_PERM) {
+				char perms[5];
+				unsigned int val = r->values[i];
+				perms[0] = 0;
+				if (val & AUDIT_PERM_READ)
+					strcat(perms, "r");
+				if (val & AUDIT_PERM_WRITE)
+					strcat(perms, "w");
+				if (val & AUDIT_PERM_EXEC)
+					strcat(perms, "x");
+				if (val & AUDIT_PERM_ATTR)
+					strcat(perms, "a");
+				if (watch) {
+					field_value_len = snprintf(
+						NULL, 0, " -p %s", perms);
+					char *audit_field =
+						malloc(field_value_len + 1);
+					snprintf(audit_field,
+						 field_value_len + 1, " -p %s",
+						 perms);
+					audit_line_push(audit_line,
+							audit_field);
+					free(audit_field);
+				} else {
+					field_value_len = snprintf(
+						NULL, 0, " -F perm=%s", perms);
+					char *audit_field =
+						malloc(field_value_len + 1);
+					snprintf(audit_field,
+						 field_value_len + 1,
+						 " -F perm=%s", perms);
+					audit_line_push(audit_line,
+							audit_field);
+					free(audit_field);
+				}
+			} else if (field == AUDIT_INODE) {
+				// This is unsigned
+				field_value_len =
+					snprintf(NULL, 0, " -F %s%s%u", name,
+						 audit_operator_to_symbol(op),
+						 r->values[i]);
+				char *audit_field = malloc(field_value_len + 1);
+				snprintf(audit_field, field_value_len + 1,
+					 " -F %s%s%u", name,
+					 audit_operator_to_symbol(op),
+					 r->values[i]);
+				audit_line_push(audit_line, audit_field);
+				free(audit_field);
+			} else if (field == AUDIT_FIELD_COMPARE) {
+				print_field_cmp(r->values[i], op, audit_line);
+			}
+			// This section is based on internal libaudit and auparse component, it's keep here for information
+			/* else if (field >= AUDIT_ARG0 && field <= AUDIT_ARG3)
+			{
+				if (field == AUDIT_ARG0)
+					a0 = r->values[i];
+				else if (field == AUDIT_ARG1)
+					a1 = r->values[i];
+
+				// Show these as hex
```

Comment:
At a minimum, you should include the above field test and the next if statement so that the arguments are converted to hex.